include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -348,6 +348,112 @@ parse_cigar(cigar_input_type && cigar_input)
     return {operations, ref_length, seq_length, sc_begin_count, sc_end_count};
 }
 
+/*!\brief Parses a cigar string into a vector of operation-count pairs (e.g. (M, 3)).
+ * \ingroup alignment_file
+ * \tparam cigar_input_type The type of a single pass input view over the cigar string; must model
+ *                          std::ranges::InputRange.
+ * \param[in]  cigar_input  The single pass input view over the cigar string to parse.
+ *
+ * \returns A tuple of size five containing (1) std::vector of operation-count pairs, e.g. (M, 3), that describe
+ *          the alignment, (2) the aligned reference length, (3) the aligned query sequence length, (4) The number of
+ *          soft clipped bases at the start of the alignment and (5) the number of bases at the end of the alignment.
+ *
+ * \details
+ *
+ * For example, the view over the cigar string "1S4M1D2M2S" will return `{[(M,4), (D,1), (M,2)], 7, 6, 1, 2}`.
+ */
+template <std::ranges::InputRange cigar_input_type>
+std::tuple<std::vector<std::pair<char, size_t>>, size_t, size_t, size_t, size_t>
+parse_binary_cigar(cigar_input_type && cigar_input, size_t n_cigar_op)
```

Comment:
> And why not passing it as uint16_t according to the specification?

:+1: 

> are you checking the limit of n_cigar_op here or somewhere else?

why should I ? If it is bigger than representable by uint16_t it will be found in the optional sam tag GC (which I have to do still), so here the range will always be correct.